### PR TITLE
Forms

### PR DESCRIPTION
**Pull request by roshan-bajgain**
### Blog app - Add forms
- Create forms to perform the following functions:
  - Creates a Post on behalf of the current_user.
  - Create a comment on behalf of the current_user.
- Allow Users to add likes to Posts.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -9,4 +9,27 @@ def show
...
+        if @new_post.save
+          redirect_to "/users/#{@new_post.author.id}/posts/", notice: 'Success!'
+        else
+          render :new, alert: 'Error occured!'
+        end
+      end
+    end
+  end
+
+  private
```
Comment (VICTOR-CHUKWU) [inline app/controllers/posts_controller.rb:30]: - [x] currently when I try creating a post I get an error of unknow variable curren_user, kindly add a method current user in your application_controller.rb. it should look like this 
```
  class ApplicationController < ActionController::Base
    def current_user
      ......
      end
end
```
Screenshot from 2022-06-08 13-55-53

File: app/views/posts/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<div class="create-post-c">
+<h2>Create a post</h2>
+
+<%= from_with model: @post, url: user_posts_path, scope: :post, html: {class: "post-form-c"} do |form| %>
```
Comment (VICTOR-CHUKWU) [inline app/views/posts/new.html.erb]: - [x] Kindly change `from_with` to `form_with` as this is causing an error when trying to view posts
    
Screenshot from 2022-06-08 14-02-28

File: config/routes.rb [hunk 3]
```diff
@@ -13,6 +13,9 @@
   # get "/users/:id/posts/:id", to: "posts#post"
 
   resources :users, only: [:show, :index] do
-    resources :posts, only: [:show, :index]
+    resources :posts, only: [:show, :index, :create, :new] do
+      resources :comments, only: [:create, :new]
+      resources :likes, only: [:create]
+      end
   end
 end
```
Comment (VICTOR-CHUKWU) [inline config/routes.rb]: - [x] [Optional]I do not know why you removed the root route, Kindly make your user index your root route

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @roshan-bajgain  👋  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Good PR
- Good commit
- All linters passed


## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by roshan-bajgain** [comment]
@VICTOR-CHUKWU  Thanks for review

**Review by ShoiraTa — APPROVED** [review]
## **STATUS: APPROVED  🟢 ** 
------
### Hi @roshan-bajgain , 

**Great work on making the changes requested by a previous reviewer 👏🏻** 

Your project is complete! . There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

5SW

**Cheers and Happy coding! :clap: :clap: :clap:**

Feel free to leave any questions or comments in the PR thread by mentioning my Github username if something is not 100% clear.
***
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have  limited number of  reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**